pkgs/tools/networking/maxscale/default.nix

```diff
@@ -0,0 +1,126 @@
+{ stdenv, fetchFromGitHub, cmake, pkgconfig, gcc, glibc
+, bison2, curl, flex, gperftools, jansson, jemalloc, kerberos, lua, mariadb
+, ncurses, openssl, pcre, pcre2, perl, rabbitmq-c, sqlite, tcl
+, libaio, libedit, libtool, libui, libuuid, zlib
+}:
+
+stdenv.mkDerivation rec {
+  name = "maxscale-${version}";
+  version = "2.1.13";
+
+  src = fetchFromGitHub {
+    owner = "mariadb-corporation";
+    repo = "MaxScale";
+    rev = "${name}";
+    sha256 = "0bwm98mbjay24vf95rk4slvjqxqblx14hvv75dpp1lhfhw25hhrf";
+  };
+
+  nativeBuildInputs = [ cmake pkgconfig ];
+
+  buildInputs = [
+    bison2 curl flex gperftools jansson jemalloc kerberos lua mariadb.connector-c
+    ncurses openssl pcre pcre2 perl rabbitmq-c sqlite tcl
+    libaio libedit libtool libui libuuid zlib
+  ];
+
+  patches = [ ./getopt.patch ];
+
+  preConfigure = ''
+    for i in `grep -l -R '#include <getopt.h>' .`; do
+      substituteInPlace $i --replace "#include <getopt.h>" "#include <${glibc.dev}/include/getopt.h>"
+    done
+ '';
+
+  cmakeFlags = [
+    "-DUSE_C99=YES"
+    "-DDEFAULT_ADMIN_USER=root"
+    "-DWITH_MAXSCALE_CNF=YES"
+    "-DSTATIC_EMBEDDED=YES"
+    "-DBUILD_RABBITMQ=YES"
+    "-DBUILD_BINLOG=YES"
+    "-DBUILD_CDC=NO"
+    "-DBUILD_MMMON=YES"
+    "-DBUILD_LUAFILTER=YES"
+    "-DLUA_LIBRARIES=${lua}/lib"
+    "-DLUA_INCLUDE_DIR=${lua}/include"
+    "-DGCOV=NO"
+    "-DWITH_SCRIPTS=OFF"
+    "-DBUILD_TESTS=NO"
+    "-DBUILD_TOOLS=NO"
+    "-DPROFILE=NO"
+    "-DWITH_TCMALLOC=YES"
+    "-DWITH_JEMALLOC=YES"
+    "-DINSTALL_EXPERIMENTAL=YES"
+    "-DTARGET_COMPONENT=all"
+  ];
+
+  CFLAGS = "-std=gnu99";
+
+  enableParallelBuilding = true;
+
+  postInstall = ''
+    patchelf --set-rpath ${stdenv.lib.makeLibraryPath [ curl gcc.cc.lib glibc jemalloc kerberos openssl pcre2 zlib ]}:${mariadb.connector-c}/lib/mariadb:$out/lib/maxscale $out/bin/dbfwchk
```

Comment:
I think this needs to be refactored somewhat before merging. I suggest lifting the `makeLibraryPath`s and use loops for common rpaths. That said I'm a little confused why all this patching is necessary to begin with?

Reply:
In the end of packet build the following message appears:
`
-- Installing: /nix/store/xgjzyvw7y1f4jh3sx1jkd46c51y5qf1y-maxscale-2.1.13/bin/maxadmin
-- Set runtime path of "/nix/store/xgjzyvw7y1f4jh3sx1jkd46c51y5qf1y-maxscale-2.1.13/bin/maxadmin" to ":/nix/store/xgjzyvw7y1f4jh3sx1jkd46c51y5qf1y-maxscale-2.1.13//nix/store/xgjzyvw7y1f4jh3sx1jkd46c51y5qf1y-maxscale-2.1.13/lib/maxscale
`
And as result, incorrect path to libmaxscale-common.so.1.0.0 library is written. Maybe there's another way to avoid this problem?